subsys/bluetooth/services/dis.c

```diff
@@ -45,6 +45,17 @@ static struct dis_pnp dis_pnp_id = {
 };
 #endif
 
+#if defined(CONFIG_BT_DIS_SYSTEM_ID)
+static uint8_t dis_system_id[8] = { (CONFIG_BT_DIS_SYSTEM_ID_OUI >> 16) & 0xFF, 
+									(CONFIG_BT_DIS_SYSTEM_ID_OUI >> 8) & 0xFF, 
+									(CONFIG_BT_DIS_SYSTEM_ID_OUI >> 0) & 0xFF,
+									((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 32) & 0xFF,
+									((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 24) & 0xFF,
+									((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 16) & 0xFF,
+									((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 8) & 0xFF,
+									((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER >> 0) & 0xFF };
```

Comment:
This is in big endian format. Is this correct?

The GATT Specification Supplement say the following:

> All fields in a characteristic or descriptor are little endian unless otherwise stated.

Reply:
They also compared it to addresses, which would have the OUI in big-endian order. But it might be incorrect. 

I found this StackOverflow post which does not have a clear answer: https://stackoverflow.com/questions/44295168/format-of-system-id-generated-based-on-a-bluetooth-device-address

Espressif does have support for this in their implementation, but only as a string. I could not find anyone using it on GitHub.

Reply:
I have now switched to using little-endian.

Reply:
I actually don't know what is correct. I did not mean to make you change it before we know. My colleagues and I have since looked at the specification and it's not at all obvious. There are reasons to think it should be big-endian.

 - The spec for System ID talks in terms of concatenation, and provides a number as an example. If we assume concatenation is a left-to-right operation on the output stream, then the example looks like big-endian.
 - The spec for System ID references the [IEEE EUI spec], which on page 10 says it's "typically" sent in big-endian.

[IEEE EUI spec]: https://standards.ieee.org/wp-content/uploads/import/documents/tutorials/eui.pdf

Reply:
Sounds like big-endian might be right then? I had only changed one place, so now the initial value will differ from values set later using runtime settings.

Reply:
Yes. Big-endian seems right. Let's just go for that.

We may want to open up a erratum request with the Bluetooth SIG and ask for a clarification in the spec. Would you be interested in doing this?

Reply:
I switched back to big-endian.

I do not know how to open requests with SIG. Seems like they at least have some contact forms behind login. I do not have credentials, so perhaps it would be easier if someone more involved with Zephyr could do it.

Reply:
I could change to the nRF51 SDK behaviour, or should we wait for a response from the SIG?

Reply:
I have no idea what the resolution will be or how long it will take. Your guess is as good as mine. I think we should add `select EXPERIMENTAL` to the kconfig until this is resolved, along with an explanation that it's experimental because the encoding is not clearly standardized.

Btw, I found another document, that looks like it's from Bluetooth SIG, but I have no idea where it is officially published. Anyway, it's also confusing, referring to "LSOs" and "MSOs" without defining these terms. https://github.com/sputnikdev/bluetooth-gatt-parser/blob/master/src/main/resources/gatt/characteristic/org.bluetooth.characteristic.system_id.xml

We could alternatively just treat this characteristic as a fixed-length byte array, without any further interpretation. This is how the Bosch API does it, by the look of it.

Reply:
It says in the bottom that

> The fields in the above table are in the order of LSO to
>  MSO. Where LSO = Least Significant Octet and MSO = Most
> Significant Octet.

My take from this document is that this indicates the same as used in nRF51.

Since we do not know how long the request will take, I would suggest that we implement the same method as in nRF51 and mark it as experimental.